addons/mail/static/tests/discuss_app/jump_to_present.test.js

```diff
@@ -260,3 +260,23 @@ test("show jump to present banner after scrolling up 10 messages", async () => {
     await scroll(".o-mail-Thread", queryFirst(".o-mail-Thread").scrollTop - 5 * messageHeight);
     await contains("[title='Jump to Present']");
 });
+
+test("focus composer after jump to present", async () => {
+    const pyEnv = await startServer();
+    const channelId = pyEnv["discuss.channel"].create({ name: "General" });
+    pyEnv["mail.message"].create(
+        [...Array(40).keys()].map((i) => ({
+            body: `<p>Non Empty Message ${i}</p>`,
+            message_type: "comment",
+            model: "discuss.channel",
+            res_id: channelId,
+        }))
+    );
+    await start();
+    await openDiscuss(channelId);
+    await scroll(".o-mail-Thread", 0);
+    queryFirst(".o-mail-Composer-input").blur();
+    await contains(".o-mail-Composer-input:not(:focus)");
+    await click("[title='Jump to Present']");
+    await contains(".o-mail-Composer-input:focus");
```

Comment:
Prefer `.o-mail-Composer.o-focused`. While this is more fake than `input:focus`, the problem of `input:focus` is that this is not a DOM mutation and can waste time to run the test, i.e. the `contains()` could be resolved only after 3 seconds timeout.

Reply:
I remember having this code but I'm not sure we have it in 18.0.